src/avs/arpa/arpa.go

```diff
@@ -46,21 +51,65 @@ type RegistrationInfo struct {
 	DKGPublicKey []byte
 }
 
-// PrepareRegistration aggregates all required info from the node operator that
-// the ether.fi admin will need to register them to the AVS
-func (a *API) PrepareRegistration(operator *etherfi.Operator, dkgPublicKey []byte) error {
+// Register is called by the node operator with their `Node Account` to register with ARPA
+func (a *API) Register(operator *etherfi.Operator, dkgPublicKey []byte, inputSignature ISignatureUtilsSignatureWithSaltAndExpiry) error {
 
-	ri := RegistrationInfo{
-		OperatorID:   operator.ID,
-		DKGPublicKey: dkgPublicKey,
+	// registration will be signed the node operator with their `Node Account`
+	signingKey, err := crypto.HexToECDSA(os.Getenv("PRIVATE_KEY"))
+	if err != nil {
+		return fmt.Errorf("invalid private key: %w", err)
+	}
+
+	// pack input for `nodeRegister` call
+	nodeRegistryABI, err := NodeRegistryMetaData.GetAbi()
```

Comment:
you should be able to replace the next ~30 lines with just `a.nodeRegistry.NodeRegister()`. 
The generated code takes care of all these steps for you

Reply:
I hate my life